File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/Visualizer.java
/**
 *
 * @file
 *
 * @brief Visual data provider interface
 *
 * @author vitamin.caig@gmail.com
 *
 */

package app.zxtune.playback;

public interface Visualizer {

  /**
   * Get currently playing spectrum
   * @param levels array of current spectrum levels playing (96 max)
   * @return count of actually stored values in bands/levels (less or equal to levels.length)
   */
  int getSpectrum(byte[] levels) throws Exception;
}

File: apps/zxtune-android/zxtune/src/main/java/app/zxtune/playback/Visualizer.kt
/**
 *
 * @file
 *
 * @brief Visual data provider interface
 *
 * @author vitamin.caig@gmail.com
 */
package app.zxtune.playback

interface Visualizer {
    /**
     * Get currently playing spectrum
     * @param levels array of current spectrum levels playing (96 max)
     * @return count of actually stored values in bands/levels (less or equal to levels.length)
     */
    @Throws(Exception::class)
    fun getSpectrum(levels: ByteArray): Int
}
